cli/command_snapshot_fix.go

```diff
@@ -0,0 +1,188 @@
+package cli
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+
+	"github.com/kopia/kopia/repo"
+	"github.com/kopia/kopia/repo/blob"
+	"github.com/kopia/kopia/repo/manifest"
+	"github.com/kopia/kopia/snapshot"
+	"github.com/kopia/kopia/snapshot/snapshotfs"
+)
+
+type commandSnapshotFix struct {
+	removeObjectIDs    []string
+	removeFilesByName  []string
+	fullRead           bool
+	removeInvalidFiles bool
+
+	// snapshots to target
+	manifestIDs []string
+	sources     []string
+
+	commit bool
+
+	verifier *snapshotfs.Verifier
+}
+
+func (c *commandSnapshotFix) setup(svc appServices, parent commandParent) {
+	cmd := parent.Command("fix", "Fix snapshot consistency issues")
+
+	// action to perform
+	cmd.Flag("remove-invalid-files", "Delete invalid files").BoolVar(&c.removeInvalidFiles)
+	cmd.Flag("remove-object", "Remove objects with specified ID").StringsVar(&c.removeObjectIDs)
+	cmd.Flag("remove-file", "Remove file with specified name (unqualified)").StringsVar(&c.removeFilesByName)
```

Comment:
Since these are mutually exclusive in usage (or I imagine so), can we make them subcommands rather than flags?

```shell
kopia snapshot fix remove-invalid-files
kopia snapshot fix remove-object
kopia snapshot fix remove-file
```

What do you think?

Reply:
That's a great idea. I implemented something very similar, PTAL.